tests/fundAccountPage.test.ts

```diff
@@ -0,0 +1,148 @@
+
+    const getLumenButton = page
+      .getByTestId("fundAccount-buttons")
+      .getByText("Get lumens");
+    const fillInButton = page
+      .getByTestId("fundAccount-buttons")
+      .getByText("Fill in with generated key");
+
+    await expect(getLumenButton).toBeDisabled();
+    await expect(fillInButton).toBeDisabled();
+  });
+
+  test("Gets an error with an invalid public key in 'Public Key' field", async ({
+    page,
+  }) => {
+    const publicKeyInput = page.locator("#generate-keypair-publickey");
+
+    // Type in an invalid string in 'Public Key' input field
+    await publicKeyInput.fill("XLKDSFJLSKDJF");
+
+    await expect(publicKeyInput).toHaveAttribute("aria-invalid", "true");
+    await expect(
+      page.getByTestId("fundAccount-buttons").getByText("Get lumens"),
+    ).toBeDisabled();
+  });
+
+  test("Successfully funds an account when clicking 'Get lumens' with a valid public key", async ({
+    page,
+  }) => {
+    // Get a new public key
+    const publicKey =
+      "GDVOT2ALMUF3G54RBHNJUEV6LOAZCQQCARHEVNUPKGMVPWFC4PFN33QR";
+    const publicKeyInput = page.locator("#generate-keypair-publickey");
+    const getLumenButton = page
+      .getByTestId("fundAccount-buttons")
+      .getByText("Get lumens");
+
+    // Type in an invalid string in 'Public Key' input field
+    await publicKeyInput.fill(publicKey);
+
+    await expect(publicKeyInput).toHaveAttribute("aria-invalid", "false");
+    await expect(getLumenButton).toBeEnabled();
+
+    // Mock the friendbot api call
+    await page.route("**/*", async (route, request) => {
+      if (request.url().includes("?addr=")) {
+        await route.fulfill({
+          status: 200,
+          contentType: "application/json",
+          body: JSON.stringify({}),
+        });
+      } else {
+        await route.continue();
+      }
+    });
+
+    // Ensure the listener is set up before the action that triggers the request
+    const responsePromise = page.waitForResponse(
+      (response) =>
+        response.url().includes("?addr=") && response.status() === 200,
+    );
+
+    await getLumenButton.click();
+
+    // Wait for the mocked response
+    await responsePromise;
+
+    // Success <Alert/> is visible
+    const alertBox = page.getByText(/Successfully funded/);
+    await expect(alertBox).toBeVisible();
+  });
+
+  test("Gets an error when submitting 'Get lumens' button twice with a valid public key", async ({
+    page,
+  }) => {
+    const publicKeyInput = page.locator("#generate-keypair-publickey");
+    const getLumenButton = page
+      .getByTestId("fundAccount-buttons")
+      .getByText("Get lumens");
+
+    // Type in an invalid string in 'Public Key' input field
+    await publicKeyInput.fill(
+      "GBX6W44ISK6XTBDDJ5ND6KTJHLYEYHMR4SDG635NRARYVJ3G2YXGDT6Y",
+    );
+
+    await expect(publicKeyInput).toHaveAttribute("aria-invalid", "false");
+    await expect(getLumenButton).toBeEnabled();
+
+    await getLumenButton.click();
```

Comment:
I could also mock the api call to return an error vs. clicking the button twice

Reply:
Yes, let's mock an error response to control this better.